subprojects/core/src/main/java/org/gradle/api/internal/artifacts/dependencies/ModuleDependencyCapabilitiesInternal.java

```diff
@@ -17,11 +17,10 @@
 
 import org.gradle.api.artifacts.ModuleDependencyCapabilitiesHandler;
 import org.gradle.api.capabilities.Capability;
-
-import java.util.List;
+import org.gradle.api.provider.SetProperty;
 
 public interface ModuleDependencyCapabilitiesInternal extends ModuleDependencyCapabilitiesHandler {
-    List<Capability> getRequestedCapabilities();
+    SetProperty<Capability> getRequestedCapabilities();
```

Comment:
Can we expose this as a `SetProvider`? This interface is not necessarily mutable, only the implementation specifies `Mutable` in the name, and all external callers of this method only read the value.

Reply:
I think you mean `Provider<Set<...>>`, but yes.